tap_klaviyo/streams.py

```diff
@@ -75,11 +85,102 @@ def post_process(
+
+
+class CampaignValuesReportsStream(KlaviyoStream):
+    name = "campaign_values_reports"
+    path = "/campaign-values-reports"
+    rest_method = "POST"
+    primary_keys = ["id"]
+    replication_key = None
+    parent_stream_type = CampaignsStream
+    schema_filepath = SCHEMAS_DIR / "campaign_values_reports.json"
+    backoff_max_tries = 10
+    state_partitioning_keys = []
+    ignore_parent_replication_key = True
+
+    def post_process(self, row: dict, context: dict | None = None) -> dict | None:
+        row["campaign_id"] = context["campaign_id"]
+        row["campaign_status"] = context["campaign_status"]
+        row["generated_at"] = datetime.now().isoformat()
+        return row
+
+    def prepare_request_payload(
+            self,
+            context: dict | None,
+            next_page_token: t.Optional[t.Any],
+    ) -> dict | None:
+        return {
+            "data": {
+                "type": "campaign-values-report",
+                "attributes": {
+                    "statistics": [
+                        "click_rate",
+                        "click_to_open_rate",
+                        "clicks",
+                        "clicks_unique",
+                        "delivered",
+                        "delivery_rate",
+                        "open_rate",
+                        "opens",
+                        "opens_unique",
+                        "recipients",
+                        "unsubscribe_rate",
+                        "unsubscribe_uniques",
+                        "unsubscribes",
+                        ],
+                    "timeframe": {
+                        "key": "last_12_months"
+                    },
+                    "conversion_metric_id": "WcGvVS",
+                    "filter": f"equals(campaign_id,\"{context['campaign_id']}\")"
+                }
+            }
+        }
+
+    def backoff_wait_generator(self) -> t.Generator[float, None, None]:
+        def _backoff_from_headers(retriable_api_error):
+            response_headers = retriable_api_error.response.headers
+            return int(response_headers.get("Retry-After", 60))
+
+        return self.backoff_runtime(value=_backoff_from_headers)
+
+    def get_records(self, context: dict | None) -> t.Iterable[dict[str, t.Any]]:
+        if not self.config.get("report_campaigns_sent_last_n_days"):
+            raise MissingConfigException('report_campaigns_sent_last_n_days config is required '
+                                         'to limit the reports requests.')
+        # Only fetch records if the campaign has been sent
+        if (context['campaign_status'] in ('Sent', 'Variations Sent')
+                and context.get('send_time')
+                # Campaigns sent in the last n days
+                and datetime.fromisoformat(context['send_time']) >=
+                datetime.now(timezone.utc) - timedelta(days=self.config['report_campaigns_sent_last_n_days'])):
```

Comment:
We will limit the number of campaigns based on the campaigns sent in the last N days. We need to limit it because this endpoint has only a few requests daily.